### PR TITLE
Create 'Projects' dynamic list

### PR DESCRIPTION
**Pull request by jcapona**
| Status  | Ticket/Issue |
| :---: | :--: |
| Ready/Hold | [Ticket](https://pi-top.atlassian.net/browse/OS-1270) |

#### Main changes
- Add `SelectableList` core component. It draws a list that the user can navigate through, highlighting the currently selected item.
- Add 'Projects' menu entry to main 'hud'.
- 'Projects' menu opens a `ProjectList`, based on `SelectableList`, with the projects found in `~/Desktop/Projects`. If no valid projects are found, a 'no projects found' message is displayed. Otherwise, the projects are displayed and users can navigate them. Newest projects are displayed first.
- When clicking the select button on a project entry, it opens a new page with the project title and image.

#### Screenshots (feature, test output, profiling, dev tools etc)

[insert screenshots here]

#### Other notes (e.g. implementation quirks, edge cases, questions / issues)
-

#### Manual testing tips
-

#### Tag anyone who definitely needs to review or help
-

### CODE REVIEW

File: pt_miniscreen/pages/root/projects.py [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+            MarqueeText,
+            text=self.project_config.title,
+            font_size=10,
+            align="center",
+            vertical_align="center",
+        )
+        self.page = partial(ProjectPage, self.project_config)
+
+    def render(self, image):
+        return apply_layers(
```
Comment (olivierwilkinson) [inline pt_miniscreen/pages/root/projects.py:102]: I think this can return self.text.render, eg

```
def render(self, image)
  return self.text.render(image)
```

I don't remember the defaults for MarqueeText but I believe this will render left horizontally aligned and centre vertically aligned in the image passed, which will be the size of the row.

File: pt_miniscreen/pages/root/projects.py [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+        self.text = self.create_child(
+            MarqueeText,
+            text="No projects found",
+            font_size=10,
+            align="center",
+            vertical_align="center",
+        )
+
+    def render(self, image):
+        return apply_layers(
```
Comment (olivierwilkinson) [inline pt_miniscreen/pages/root/projects.py:127]: Ditto here for rendering self.text.render

File: pt_miniscreen/pages/root/projects.py [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+
+    def __init__(self, **kwargs) -> None:
+        super().__init__(
+            Rows=self.load_project_rows(),
+            num_visible_rows=5,
+            **kwargs,
+        )
+
+    def load_project_rows(self) -> List:
+        pages: List[Union[Type[EmptyProjectRow], partial[ProjectRow]]] = []
```
Comment (olivierwilkinson) [inline pt_miniscreen/pages/root/projects.py:152]: I think this variable should be called rows rather than pages (picky of me sorry)

**Review by olivierwilkinson — COMMENTED** [review]
Looks great! I have a couple minor comments but they don't block merging at all.

Great test suite!

I haven't explicitly approved only because I haven't run it on my pi-top 👍

File: pt_miniscreen/pages/root/projects.py (hunk 1 above)
Comment (jcapona) [reply pt_miniscreen/pages/root/projects.py:102]: good one! the image changed a bit (the text shifted 1 pixel based on the test snapshots) but I prefer the simplicity of the new render method
